akka-http-core/src/test/scala/akka/http/scaladsl/ClientServerSpec.scala

```diff
@@ -694,6 +698,17 @@ class ClientServerSpec extends WordSpec with Matchers with BeforeAndAfterAll wit
         connSourceSub.cancel()
       }
     }
+
+    "produce a useful error message when connecting to a HTTP endpoint over HTTPS" in Utils.assertAllStagesStopped {
+      val dummyFlow = Flow.fromFunction((_: HttpRequest) ⇒ ???)
+
+      val binding = Http().bindAndHandle(dummyFlow, "127.0.0.1", port = 0).futureValue
+      val uri = "https://" + binding.localAddress.getHostString + ":" + binding.localAddress.getPort
+
+      EventFilter.warning(pattern = "Perhaps this was an HTTPS request sent to an HTTP endpoint", occurrences = 6) intercept {
+        Await.ready(Http().singleRequest(HttpRequest(uri = uri)), 30.seconds)
+      }
+    }
```

Comment:
Did you remove the `unbind` from the previous version accidentally or by purpose?

Reply:
oops, no (removed it when trying to use `TestSetup`, but then decided this is clearer and forgot to put it back)